pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2786,7 +2786,9 @@ private void receiveRFQ() throws IOException {
   }
 
   @Override
+  @SuppressWarnings("deprecation")
   protected void sendCloseMessage() throws IOException {
+    LOGGER.log(Level.FINEST, " FE=> Terminate");
     pgStream.sendChar('X');
     pgStream.sendInteger4(4);
   }
```

Comment:
this method appears to only (previously) be called from QueryExecutorBase.close, which is refactored in this PR to get rid of. 
should this just be removed?

Reply:
Method removal breaks backward compatibility which is undesired

Reply:
should it at least be deprecated then?
It is a protected method on a class with no descendants in this project. Given that we have moved the behavior into the finalize action, there does not seem to be any reasonable usage of this method being called directly.

Reply:
I deprecated `sendCloseMessage` in the base class

Reply:
I've moved `sendCloseMessage` to `QueryExecutorCloseAction` so reduce code duplication